### PR TITLE
[WIP] Expand without origin

### PR DESCRIPTION
**Pull request by stegben**
Yoctol/intent-classifier-service#805

### CODE REVIEW

File: uttut/tests/test_expand_by_entities.py [hunk 1]
```diff
@@ -16,11 +17,17 @@
...
     def setUp(self):
+        self.gen_old_data()
+        self.gen_new_data()
+
+    def gen_old_data(self):
         self.utterance = '我想訂明天從紐約飛到新加坡的機票'
         self.entities = [
-            Entity(name='日期', value='明天', start=3, end=5, replacements=['下禮拜二']),
+            Entity(name='日期', value='明天', start=3,
+                   end=5, replacements=['下禮拜二']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 2]
```diff
@@ -16,11 +17,17 @@
...
             Entity(name='出發地', value='紐約', start=6, end=8),
-            Entity(name='目的地', value='新加坡', start=10, end=13, replacements=['斯堪地那維亞', 'KIX']),
+            Entity(name='目的地', value='新加坡', start=10,
+                   end=13, replacements=['斯堪地那維亞', 'KIX']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 3]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
         ]
         self.expected_entity_lists = [
             [
-                Entity(name='日期', value='明天', start=3, end=5, replacements=['下禮拜二']),
+                Entity(name='日期', value='明天', start=3,
+                       end=5, replacements=['下禮拜二']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 4]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
...
                 Entity(name='出發地', value='紐約', start=6, end=8),
-                Entity(name='目的地', value='新加坡', start=10, end=13, replacements=['斯堪地那維亞', 'KIX']),
+                Entity(name='目的地', value='新加坡', start=10,
+                       end=13, replacements=['斯堪地那維亞', 'KIX']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 5]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
...
             ],
             [
-                Entity(name='日期', value='下禮拜二', start=3, end=7, replacements=['明天']),
+                Entity(name='日期', value='下禮拜二', start=3,
+                       end=7, replacements=['明天']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 6]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
...
                 Entity(name='出發地', value='紐約', start=8, end=10),
-                Entity(name='目的地', value='新加坡', start=12, end=15, replacements=['斯堪地那維亞', 'KIX']),
+                Entity(name='目的地', value='新加坡', start=12,
+                       end=15, replacements=['斯堪地那維亞', 'KIX']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: Avoid coding style that is inconsistent with our repo

File: uttut/tests/test_expand_by_entities.py [hunk 7]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
...
             ],
             [
-                Entity(name='日期', value='明天', start=3, end=5, replacements=['下禮拜二']),
+                Entity(name='日期', value='明天', start=3,
+                       end=5, replacements=['下禮拜二']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: same

File: uttut/tests/test_expand_by_entities.py [hunk 8]
```diff
@@ -41,34 +48,46 @@ def setUp(self):
...
+                       end=13, replacements=['斯堪地那維亞', '新加坡']),
             ],
             [
-                Entity(name='日期', value='下禮拜二', start=3, end=7, replacements=['明天']),
+                Entity(name='日期', value='下禮拜二', start=3,
+                       end=7, replacements=['明天']),
                 Entity(name='出發地', value='紐約', start=8, end=10),
-                Entity(name='目的地', value='KIX', start=12, end=15, replacements=['新加坡', '斯堪地那維亞']),
+                Entity(name='目的地', value='KIX', start=12,
+                       end=15, replacements=['新加坡', '斯堪地那維亞']),
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: too much inconsistent coding style

File: uttut/tests/test_expand_by_entities.py [hunk 9]
```diff
@@ -114,11 +133,11 @@ def setUp(self):
...
-                self.expected_utterances,
-                self.expected_entity_lists,
-                self.expected_entity_wo_replacements_lists,
-                self.expected_intents,
-            ):
+            self.expected_utterances,
+            self.expected_entity_lists,
+            self.expected_entity_wo_replacements_lists,
+            self.expected_intents,
+        ):
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: why?

Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 8 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 7 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 6 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 5 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 4 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 3 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py (hunk 1 above)
Comment (SoluMilken) [reply uttut/tests/test_expand_by_entities.py]: e1ac824

File: uttut/tests/test_expand_by_entities.py [hunk 10]
```diff
@@ -176,3 +258,30 @@ def test_partition_by_entities(self):
             parts,
             ['我想訂', '明天', '從', '紐約', '飛到', '新加坡', '的機票'],
         )
+
+    def test_expand_without_origin(self):
+        utterance_with_tag = '我想訂<日期>從<出發地>飛到<目的地>的機票'
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: 何不用 `self.utterance_with_tag?`

File: uttut/tests/test_expand_by_entities.py [hunk 11]
```diff
@@ -176,3 +258,30 @@ def test_partition_by_entities(self):
...
+            Entity(name='出發地', value='<出發地>', start=8, end=13),
+            Entity(
+                name='目的地', value='<目的地>', start=15, end=20, replacements=['新加坡', '斯堪地那維亞', 'KIX']),
+        ]
+        intents = [Intent(name='訂機票')]
+        datum = Datum(
+            utterance=utterance_with_tag,
+            intents=intents,
+            entities=entities,
+        )
```
Comment (stegben) [inline uttut/tests/test_expand_by_entities.py]: 這和 `self.datum_with_tag` 有什麼差異？

**Comment by stegben** [comment]
Use #11

**Pull request closed without merging**